### DISCUSSION TITLE
How to work with client in multiple files?

### DISCUSSION THREAD

**Discussion by Arctomachine** [prisma/prisma#20371] [Q&A] [answered]
If my workflow requires to create multiple files, one for each model in database, then how to handle connection via client?

Is it fine to just create new instance in each file?
```js
// ./modelA.js
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

// ./modelB.js
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```

Or is it better to create singleton/global for client and import this wrapper?

**Comment by ludralph** [comment, marked as answer]
Hi @Arctomachine 👋 

I think it's better to create a single instance of the Prisma client and then import and reuse that instance in each of your model files. Creating a singleton or a global instance for the Prisma client ensures that you have only one connection to the database throughout the application's lifecycle, which is more efficient and reduces the overhead of establishing multiple connections. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Arctomachine** [reply]
Is there guide available on creating it in both oop and procedural ways?

**Reply by ludralph** [reply]
Hi @Arctomachine, there is no guide in creating it in either oop or procedural way but in object-oriented programming approach, you can create classes to represent your models, and each class can have methods to interact with the corresponding table in the database using Prisma. For example:
```js
// modelA.js
import { PrismaClient } from '@prisma/client';

class ModelA {
  constructor() {
    this.prisma = new PrismaClient();
  }

  async findAll() {
    return this.prisma.modelA.findMany();
  }

  async create(data) {
    return this.prisma.modelA.create({ data });
  }

  async update(id, data) {
    return this.prisma.modelA.update({ where: { id }, data });
  }

  async delete(id) {
    return this.prisma.modelA.delete({ where: { id } });
  }
}

export default new ModelA();
```

In a procedural approach, you can organize your code into functions that interact with the Prisma client directly. for example:
```js
// modelA.js
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function findAll() {
  return prisma.modelA.findMany();
}

async function create(data) {
  return prisma.modelA.create({ data });
}

async function update(id, data) {
  return prisma.modelA.update({ where: { id }, data });
}

async function remove(id) {
  return prisma.modelA.delete({ where: { id } });
}

export { findAll, create, update, remove };
```

Regardless of the programming style you choose, the core interaction with Prisma remains the same. You define your database models using Prisma's schema definition language, and then you can use Prisma Client's generated methods to perform CRUD operations on the database.
